pelican/contents.py

```diff
@@ -174,10 +171,35 @@ def valid_save_as(self):
         try:
             sanitised_join(output_path, self.save_as)
         except RuntimeError:  # outside output_dir
+            logger.error(
+                "Skipping %s: file %r would be written outside output path",
+                self,
+                self.save_as,
+            )
             return False
 
         return True
 
+    def _valid_status(self):
+        if hasattr(self, 'allowed_status'):
+            if self.status not in self.allowed_status:
+                logger.error(
+                    "Unknown status '%s' for file %s, skipping it.",
+                    self.status,
+                    self
+                )
+                return False
+
+        # if undefined we allow all
+        return True
+
+    def valid(self):
```

Comment:
I would lean towards calling this method `is_valid` (so we get `article.is_valid()`).